src/cuga/backend/agent_spawn/runtime.py

```
@@ -0,0 +1,302 @@
+def pop_spawn_future(thread_id: str, future_id: str) -> None:
+    store = _futures_by_thread.get(thread_id or "_default")
+    if store is not None:
+        store.pop(future_id, None)
+        if not store:
+            _futures_by_thread.pop(thread_id or "_default", None)
+
+
+def clear_runtime_caches(thread_id: Optional[str] = None) -> None:
+    """Clear spawn futures (and cancel tracked async tasks) for one thread or all."""
+    if thread_id is None:
+        for tid, tasks in list(_tasks_by_thread.items()):
```

Ruff (0.15.20)

[warning] Loop control variable `tid` not used within loop body

Rename unused `tid` to `_tid`

(B007)